extensions/amp-analytics/0.1/events.js

```diff
@@ -1384,15 +1411,25 @@ export class VideoEventTracker extends EventTracker {
         event.target,
         'No target specified by video session event.'
       );
-      targetReady.then((target) => {
-        if (!target.contains(el)) {
-          return;
-        }
-        const normalizedDetails = removeInternalVars(details);
-        listener(new AnalyticsEvent(target, normalizedType, normalizedDetails));
+
+      Promise.all(targetReadyPromises).then((targetReady) => {
+        targetReady.forEach((target) => {
+          if (!target[0].contains(el)) {
+            return;
+          }
+          const normalizedDetails = removeInternalVars(details);
+          listener(
+            new AnalyticsEvent(target[0], normalizedType, normalizedDetails)
+          );
+        });
```

Comment:
As per my other comment, the use of `target[0]` is fishy. You don't need multiple promises once you fix the 2d array issue.

```suggestion
      Promise.all(targetPromises).then((targets) => {
        targets.forEach((target) => {
          if (!target.contains(el)) {
            return;
          }
          const normalizedDetails = removeInternalVars(details);
          listener(
            new AnalyticsEvent(target, normalizedType, normalizedDetails)
          );
        });
```